sonarts-core/src/rules/noRedundantBooleanRule.ts

```diff
@@ -44,17 +44,23 @@ export class Rule extends tslint.Rules.TypedRule {
 
 class Visitor extends SonarRuleVisitor {
   public visitBinaryExpression(node: ts.BinaryExpression) {
-    if (
-      is(
-        node.operatorToken,
-        ts.SyntaxKind.EqualsEqualsToken,
-        ts.SyntaxKind.ExclamationEqualsToken,
-        ts.SyntaxKind.BarBarToken,
-        ts.SyntaxKind.AmpersandAmpersandToken,
-      )
-    ) {
-      this.check(node.left);
-      this.check(node.right);
+    if (is(node.operatorToken, ts.SyntaxKind.BarBarToken) && node.right.getText() === "false") {
```

Comment:
why not `is(node.right, ts.SyntaxKind.FalseKeyword)`;

Reply:
indeed

Reply:
you can actually merge the ifs, and perhaps extract first condition to a small function `isOrFalse`